src/library/coverartlabel.cpp

```diff
@@ -0,0 +1,88 @@
+#include "library/coverartlabel.h"
+
+#include <QtDebug>
+
+#include "library/coverartutils.h"
+#include "library/dlgcoverartfetchedfullsize.h"
+#include "library/dlgcoverartfullsize.h"
+#include "track/track.h"
+#include "widget/wcoverartmenu.h"
+
+namespace {
+
+constexpr QSize kDeviceIndependentCoverLabelSize = QSize(100, 100);
+
+inline QPixmap scaleCoverLabel(
+        QWidget* parent,
+        QPixmap pixmap) {
+    const auto devicePixelRatioF = parent->devicePixelRatioF();
+    pixmap.setDevicePixelRatio(devicePixelRatioF);
+    return pixmap.scaled(
+            kDeviceIndependentCoverLabelSize * devicePixelRatioF,
+            Qt::KeepAspectRatio,
+            Qt::SmoothTransformation);
+}
+
+QPixmap createDefaultCover(QWidget* parent) {
+    auto defaultCover = QPixmap(CoverArtUtils::defaultCoverLocation());
+    return scaleCoverLabel(parent, defaultCover);
+}
+
+} // anonymous namespace
+
+CoverArtLabel::CoverArtLabel(QWidget* parent)
```

Comment:
Since this is still a widget, I had expect to find this is the widget folder. 
We two names, one for the base without edit menu and one for the parent class with edit menu. 

How about 
WCoverArtLabelBase and WCoverArtLabelWithEditMenu

Or 

WCoverArtDisplay and WCoverArtDisplayEditable 


or such

@ronso0 what do yo think

Reply:
Only the menu is the difference, but I can not be sure about the future steps. Maybe later, just for "Fetched Cover Art" there could be a menu such as "apply this cover" or "find the album" etc.

> Though, as I said this is not high prio IMO and can be adressed later on.

:+1: